state-chain/pallets/cf-flip/src/on_charge_transaction.rs

```diff
@@ -89,3 +117,71 @@ impl<T: TxConfig + FlipConfig + Config> OnChargeTransaction<T> for FlipTransacti
 		Ok(())
 	}
 }
+
+/// Converts a call into a call index to allow it to be categorised for fee scaling.
+pub trait CallIndexer<Call> {
+	type CallIndex: Encode + MaxEncodedLen;
+
+	fn call_index(_call: &Call) -> Option<Self::CallIndex>;
+}
+
+impl<Call> CallIndexer<Call> for () {
+	type CallIndex = ();
+
+	fn call_index(_call: &Call) -> Option<Self::CallIndex> {
+		None
+	}
+}
+
+#[derive(
+	Encode, Decode, TypeInfo, MaxEncodedLen, Clone, Copy, PartialEq, Eq, RuntimeDebug, Default,
+)]
+pub enum FeeScalingRateConfig {
+	/// No scaling for the first `threshold` calls, then scale by `exp_base`^(`call_count` -
+	/// `threshold`).
+	DelayedExponential { threshold: u16, exponent: u16 },
+	#[default]
+	NoScaling,
+}
+
+impl FeeScalingRateConfig {
+	pub fn scale_fee<Balance: AtLeast32BitUnsigned + Copy>(
+		&self,
+		pre_scaled_fee: Balance,
+		call_count: u16,
+	) -> Balance {
+		match self {
+			FeeScalingRateConfig::DelayedExponential { threshold, exponent } => {
+				let multiplier =
+					1 + call_count.saturating_sub(*threshold).saturating_pow(*exponent as u32);
+				pre_scaled_fee.saturating_mul(multiplier.into())
```

Comment:
In the base case, exponent is 1. It's the first call, so we pass through call_count = 1. Threshold is 0. So now on the first call we do 1 + 1^1 so the multiplier is 2 not one. The first call should have a multiplier of 1.

Reply:
Yeah I think I confused myself there... will fix.